includes/api/class-wc-rest-customers-controller.php

```diff
@@ -38,7 +39,7 @@ protected function get_formatted_item_data( $object ) {
 
 		// Format date values.
 		foreach ( $format_date as $key ) {
-			$datetime              = $data[ $key ];
+			$datetime              = 'date_created' == $key ? get_date_from_gmt( $data[ $key ] ) : $data[ $key ];
```

Comment:
After creating a user, if I try to get its data using the WC API, below is what I get for the `date_created` and `date_created_gmt` fields:

```
"date_created": "1970-01-01T00:00:00",
"date_created_gmt": "1970-01-01T00:00:00",
```

The value stored in the `user_registered` field is `2019-01-10 12:35:09`.

I believe this is happening because `get_date_from_gmt()` expects a string and `$data[$key]` is a instance of `WC_DateTime`. The unit test pass because the same problem happens when defining `$date_created` and `get_date_from_gmt()` also returns `1970-01-01T00:00:00`.

Besides the above, I would suggest adding a inline comment here explaining why it is necessary to use different logics to deal with `date_created` and `date_modified` (basically what you mentioned in the PR description).

Reply:
I'm getting the correct dates. When I wrote this I had PHP 7.2 but have since updated to 7.3. Are you working with PHP 5.X ?

```
  "id": 8,
  "date_created": "2019-01-10T14:56:50",
  "date_created_gmt": "2019-01-10T18:56:50",
  "date_modified": "2019-01-10T14:56:50",
  "date_modified_gmt": "2019-01-10T18:56:50",
```

Reply:
@rodrigoprimo thanks for looking at this. I have updated the PR with both suggestions.

Reply:
> When I wrote this I had PHP 7.2 but have since updated to 7.3. Are you working with PHP 5.X ?

I'm running PHP 7.2. I'm not sure why we are getting different results. I tried your newest patch and still got the wrong dates. I didn't have much time to investigate this today, but I did a quick test and it seems that on Travis the dates are also wrong:

https://travis-ci.org/rodrigoprimo/woocommerce/jobs/478494434#L662
https://github.com/rodrigoprimo/woocommerce/pull/23
https://github.com/rodrigoprimo/woocommerce/pull/23/commits/ba0d6fe3c599b8f56235513c8103ca093163f4ef#diff-def46ca6e3bffed618e410eaad72eff1R61

I can take a closer look at this next week if necessary.

Reply:
@rodrigoprimo `get_date_from_gmt( WC_Customer::get_date_created() )` always returns Jan 1, 1970 when the `timezone_string` is not set so this test has always been comparing Jan 1, 1970 to Jan 1, 1970 for the creation date.

I have added another block of code which sets the TZ to `America/New York` and checks that the API returns the correct dates.